Log cached WooCommerce version along track events

iamgabrielma opened the pull request:
## Description
We have found ourselves a few times in iOS with lack of data when trying to check WooCommerce versions from the track events dashboard, for example, recently when deciding if would be good enough to [increase the threshold of a feature from 9.6 to 9.7](https://github.com/woocommerce/woocommerce-ios/pull/16463), since we could only rely on Android data and guess iOS user base distribution would be similar

This PR mimics [Android's approach](https://github.com/woocommerce/woocommerce-android/blob/7e4005451123b7cbf41e796bcafb8e8d5844756b/WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsTracker.kt#L340) to pass the WooCommerce version as default property along track events. 

For this we retrieve, store in memory, and pass the woo version through the `cached_woo_core_version` property (for consistency with Android naming) to all track events, following the same pattern as others like `isWpcomStore`.  

## Test Steps
- With 2 different woocommerce stores, on different woocommerce versions (or test with the same store by updating the version)
- Run the app, note in the console how `cached_woo_core_version` tracks the woo version
- In menu > change store, note the version tracked is correct.

Store 1:
```
🔵 Tracked ... cached_woo_core_version: 10.3.6 ... blog_id: 250994203]
```
Store 2:
```
🔵 Tracked ... cached_woo_core_version: 10.4.2 ... blog_id: 215063064]
```


## Screenshots

N/A

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

iamgabrielma committed:
retrieve, store, and pass cachedWooCommerceVersion to track events

iamgabrielma committed:
check version only on session switch, avoid crsh by accessing to CD too many times

iamgabrielma committed:
update mock conformance

iamgabrielma committed:
add non-nil mocked cachedWooCommerceVersion

joshheald reviewed (approved):
Looks good and helpful, thanks for adding it.

FYI I struggled to test the caching – changing woo versions caused lots of other problems on my test sites and I'm a bit short of time to go deep on it. If you've seen it working, I'm sure it's fine, and the risk is low anyway.

iamgabrielma committed:
Merge branch 'trunk' into task/HACK-add-woo-cached-version-to-tracks

# Conflicts:
#	WooCommerce/Classes/Analytics/WooAnalytics.swift

iamgabrielma committed:
restore missing prop after merge conflict

the analytics key for cachedWooCommerceVersionKey was removed on c5bbc3e841cd3c3ae3f63edf742c7be40a3aff47 when resolving a merge conflict.
